pinot-query-planner/src/main/java/org/apache/pinot/query/planner/physical/colocated/GreedyShuffleRewriteVisitor.java

```diff
@@ -251,10 +250,19 @@ public Set<ColocationKey> visitWindow(WindowNode node, GreedyShuffleRewriteConte
     return node.getInputs().get(0).visit(this, context);
   }
 
+  @Override
+  public Set<ColocationKey> visitSetOp(SetOpNode setOpNode, GreedyShuffleRewriteContext context) {
+    Set<ColocationKey> colocationKeys = new HashSet<>();
```

Comment:
I think we should return an empty set here for now since the logic to auto-compute the colocation key here is a bit tricky.

Example if we do: `select col1, col2 from tbl1 union select col1, col2 from tbl2`, and the left side has a colocation key but the right side doesn't, then the result-set wouldn't have any colocation key.

Either ways we'll rewrite the colocation logic in H1, but best to be correct meanwhile as well.

Reply:
ack